src/LuaTransformer.ts

```
@@ -1631,7 +1686,7 @@ export class LuaTransformer {
             : undefined;
         const [params, dotsLiteral, restParamName] = this.transformParameters(functionDeclaration.parameters, context);
 
-        const name = this.transformIdentifier(functionDeclaration.name);
+        const name = this.transformIdentifier(functionDeclaration.name!);
```

We should probably throw an error if `name` is undefined.